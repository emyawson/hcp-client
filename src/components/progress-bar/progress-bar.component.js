import React from 'react';

import {
  ProgressBarBackground,
  ProgressBarContainer,
  ProgressBarFill,
} from './progress-bar.style';

export const ProgressBar = ({ fill, color }) => (
  <ProgressBarContainer>
    <ProgressBarBackground>
      <ProgressBarFill fill={fill || 0} color={color} />
    </ProgressBarBackground>
  </ProgressBarContainer>
);
