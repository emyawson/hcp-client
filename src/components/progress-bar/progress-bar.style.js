import styled from 'styled-components';

import { colors, borderRadius } from 'src/core';

export const ProgressBarContainer = styled.div`
  width: 100%;
`;

export const ProgressBarBackground = styled.div`
  width: 100%;
  height: 3px;
  background-color: ${colors.grayLight};
  border-radius: ${borderRadius.three};
  overflow: hidden;
`;

export const ProgressBarFill = styled.div`
  width: ${props =>
    `${props.fill > 100 ? 100 : props.fill < 0 ? 0 : props.fill}%`};
  transition: all 1000ms ease-in-out;
  height: 100%;
  background-color: ${props => `${props.color}`};
`;
