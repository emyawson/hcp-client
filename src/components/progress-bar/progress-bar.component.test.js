import React from 'react';
import { shallow } from 'enzyme';

import { ProgressBar } from './progress-bar.component';

describe('Progress bar component', () => {
  it('renders correctly', () => {
    const mockProps = {
      fill: 80,
      color: '#CF021B',
    };
    const wrapper = shallow(<ProgressBar {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders default fill as 0', () => {
    const wrapper = shallow(<ProgressBar color="#CF021B" />);
    expect(wrapper).toMatchSnapshot();
  });
});
