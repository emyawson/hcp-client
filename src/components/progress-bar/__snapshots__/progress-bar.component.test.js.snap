// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress bar component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    color="#CF021B"
    fill={80}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.div
          color="#CF021B"
          fill={80}
        />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          color="#CF021B"
          fill={80}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "color": "#CF021B",
          "fill": 80,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div
            color="#CF021B"
            fill={80}
          />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div
            color="#CF021B"
            fill={80}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "color": "#CF021B",
            "fill": 80,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Progress bar component renders default fill as 0 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    color="#CF021B"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.div
          color="#CF021B"
          fill={0}
        />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          color="#CF021B"
          fill={0}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "color": "#CF021B",
          "fill": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div
            color="#CF021B"
            fill={0}
          />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div
            color="#CF021B"
            fill={0}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "color": "#CF021B",
            "fill": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
